pandas/core/reshape/concat.py

```diff
@@ -519,6 +522,14 @@ def __init__(
         self.keys = keys
         self.names = names or getattr(keys, "names", None)
         self.levels = levels
+
+        if not is_bool(sort):
+            warnings.warn(
```

Comment:
umm we don't allow this anywhere else; why can't we just do a validate_bool on this?

Reply:
Since None was the default in the past and was not explicitly deprecated I would go with deprecation and remove in 2.0